Content.IntegrationTests/Tests/GameObjects/Components/EntityPrototypeComponentsTest.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Content.Client;
+using NUnit.Framework;
+using Robust.Shared.ContentPack;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Utility;
+using YamlDotNet.RepresentationModel;
+
+namespace Content.IntegrationTests.Tests.GameObjects.Components
+{
+    [TestFixture]
+    [TestOf(typeof(IgnoredComponents))]
+    [TestOf(typeof(Server.IgnoredComponents))]
+    public class EntityPrototypeComponentsTest : ContentIntegrationTest
+    {
+        [Test]
+        public async Task Test()
+        {
+            var (client, server) = await StartConnectedServerDummyTickerClientPair();
+
+            await server.WaitIdleAsync();
+
+            var sResourceManager = server.ResolveDependency<IResourceManager>();
+            var prototypePath = new ResourcePath("/Prototypes/");
+            var paths = sResourceManager.ContentFindFiles(prototypePath)
+                .ToList()
+                .AsParallel()
+                .Where(filePath => filePath.Extension == "yml" &&
+                                   !filePath.Filename.StartsWith("."))
+                .ToArray();
```

Comment:
Couldn't you use the IPrototypeManager for this?

Reply:
In what way? I don't think it stores any information about unknown prototypes, just skips them in EntityPrototype.